backend/coins/btc/maketx/txsize_internal_test.go

```diff
@@ -57,6 +59,51 @@ func makeSig() types.Signature {
 	return sig
 }
 
+// signatureScript returns the signature script (and witness) needed to spend from this address.
+func signatureScript(
+	t *testing.T,
+	address *addresses.AccountAddress,
+	signature types.Signature,
+) ([]byte, wire.TxWitness) {
+	t.Helper()
+	publicKey := address.PublicKey
+	switch address.AccountConfiguration.ScriptType() {
+	case signing.ScriptTypeP2PKH:
+		signatureScript, err := txscript.NewScriptBuilder().
+			AddData(append(signature.SerializeDER(), byte(txscript.SigHashAll))).
+			AddData(publicKey.SerializeCompressed()).
+			Script()
+		require.NoError(t, err)
+		return signatureScript, nil
+	case signing.ScriptTypeP2WPKHP2SH:
+		signatureScript, err := txscript.NewScriptBuilder().
+			AddData(address.RedeemScript).
+			Script()
+		require.NoError(t, err)
+		txWitness := wire.TxWitness{
+			append(signature.SerializeDER(), byte(txscript.SigHashAll)),
+			publicKey.SerializeCompressed(),
+		}
+		return signatureScript, txWitness
+	case signing.ScriptTypeP2WPKH:
+		txWitness := wire.TxWitness{
+			append(signature.SerializeDER(), byte(txscript.SigHashAll)),
+			publicKey.SerializeCompressed(),
+		}
+		return []byte{}, txWitness
+	case signing.ScriptTypeP2TR:
+		// We assume SIGHASH_DEFAULT, which defaults to SIGHASH_ALL without needing to explicitly
+		// append it to the signature. See:
+		// https://github.com/bitcoin/bips/blob/97e02b2223b21753acefa813a4e59dbb6e849e77/bip-0341.mediawiki#taproot-key-path-spending-signature-validation
+		txWitness := wire.TxWitness{
+			signature.SerializeCompact(),
+		}
+		return []byte{}, txWitness
+	default:
+		panic("Unrecognized address type.")
+	}
```

Comment:
I know we have different opinions on this so feel free to ignore this comment :D 

I personally like when the switch sets up the return values and then the return itself is outside the switch.

So something like

```golang
var signatureScript []byte
var txWitness wire.TxWitness
swtich {
// Everything the same without returns
}

return signatureScript, txWitness
```

(Or, if you want to save lines of code, you can even use named returns in the function signature :D)